### ISSUE TITLE
[Feature Request] Generic preprocessing before running algorithm

### ISSUE THREAD

**Issue by bartvanb** [vantage6/vantage6#777]
**Problem description**
The algorithm wrappers currently read the data from the database and feed it to the algorithm. Each algorithm is then responsible for implementing their own data processing, such as data transformation or filtering. It would be convenient if the wrapper have some simple, often-used preprocessing steps available that are generically useful for algorithms.

**Desired solution**
In creating tasks, there will be one or more fields looking something like
```
preprocessing={
  'transform': {
    'type': 'years-since'
    'input_column_name': 'birth_date',
    'output_column': 'age'
  },
  'filter': {
    'column_name': 'age',
    'min': 50
  },
}
```

In the example above, the intention is to calculate the age from the `birth_date` column and then filtering it to be at least 50.

**Comment by harmbuisman** [comment]
To consider:
* need type casting between input from e.g. a UI (string?) and the type expected by the function arguments, should be handled automatically such that we ideally do not have to manage it on the level of the individual preprocessing functions

**Comment by bartvanb** [comment]
Depends on #943 - it doesn't make sense implement preprocessing first as the implementation would change a lot after chaining is implemented.

**Comment by bartvanb** [comment]
This is no longer relevant, as v5.0 will introduce sessions, where preprocessing will be done by dedicated algorithms and thus not be part of the wrapper
